filebeat/prospector/log/prospector.go

```diff
@@ -632,5 +653,11 @@ func (p *Prospector) Stop() {
 	// Stop all harvesters
 	// In case the beatDone channel is closed, this will not wait for completion
 	// Otherwise Stop will wait until output is complete
-	p.registry.Stop()
+	p.harvesters.Stop()
+
+	// close state updater
+	p.stateOutlet.Close()
```

Comment:
Is the comment on line 654 still valid. It is important for reloading, that `Stop` by default waits for completion of sending all events.